libparsec/crates/client/src/certificates/workspace_list.rs

```diff
@@ -0,0 +1,104 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::collections::HashMap;
+
+use libparsec_client_connection::ConnectionError;
+use libparsec_platform_storage::certificates::{GetCertificateError, UpTo};
+use libparsec_protocol::authenticated_cmds;
+use libparsec_types::prelude::*;
+
+use super::{store::CertificateStorageOperationError, CertificatesOps, InvalidCertificateError};
+use crate::EventTooMuchDriftWithServerClock;
+
+#[derive(Debug, thiserror::Error)]
+pub enum CertificatesListSelfWorkspacesError {
+    #[error("Cannot reach the server")]
+    Offline,
+    #[error("Component has stopped")]
+    Stopped,
+    #[error(transparent)]
+    InvalidCertificate(#[from] InvalidCertificateError),
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+impl From<ConnectionError> for CertificatesListSelfWorkspacesError {
+    fn from(value: ConnectionError) -> Self {
+        match value {
+            ConnectionError::NoResponse(_) => Self::Offline,
+            // TODO: handle organization expired and user revoked here ?
+            err => Self::Internal(err.into()),
+        }
+    }
+}
+
+impl From<CertificateStorageOperationError> for CertificatesListSelfWorkspacesError {
+    fn from(value: CertificateStorageOperationError) -> Self {
+        match value {
+            CertificateStorageOperationError::Stopped => Self::Stopped,
+            CertificateStorageOperationError::Internal(err) => err.into(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum CertificatesWorkspaceInfoName {
+    Present{
```

Comment:
```suggestion
    Present {
```
In case of fmt doesn't work because disabled